x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/TransportInferenceAction.java

```diff
@@ -158,5 +149,24 @@ private ActionListener<InferenceServiceResults> createListener(
             });
         }
         return listener.delegateFailureAndWrap((l, inferenceResults) -> l.onResponse(new InferenceAction.Response(inferenceResults)));
-    };
+    }
+
+    private static ElasticsearchStatusException unknownServiceException(String service, String inferenceId) {
+        return new ElasticsearchStatusException(
+            "Unknown service [{}] for model [{}]. ",
+            RestStatus.INTERNAL_SERVER_ERROR,
```

Comment:
A request with an unknown service sounds like a bad request, not an internal server error.

Reply:
++ good catch